komodo-shell/src/main/java/org/komodo/shell/WorkspaceStatusImpl.java

```diff
@@ -692,13 +791,16 @@ public void setProvidedProperties( final Properties props ) throws Exception {
      * @see org.komodo.shell.api.WorkspaceStatus#setStateProperty(java.lang.String, java.lang.String)
      */
     @Override
-    public void setProvidedProperty(String name,
-                                    String value) {
+    public void setProvidedGlobalProperty(String name,
+                                          String value,
+                                          String valueType) {
         ArgCheck.isNotEmpty( name, "name" ); //$NON-NLS-1$
         if ( StringUtils.isEmpty( value ) ) {
             this.wsProperties.remove(name);
+            this.providedGlobalPropertyTypes.remove(name);
         } else {
             this.wsProperties.setProperty( name, value );
+            this.providedGlobalPropertyTypes.put( name, valueType );
```

Comment:
Should this handle an empty `valueType` or should it be an error? In `initProperties` you use `String` when there is an empty type. Either way I would document it in the API.